src/test/ui/in-band-lifetimes/no_introducing_in_band_in_locals.rs

```diff
@@ -7,7 +7,7 @@ fn foo(x: &u32) {
 
 fn foo2(x: &u32) {}
 fn bar() {
-    let y: fn(&'test u32) = foo2; //~ ERROR use of undeclared lifetime
+    let y: fn(&'test u32) = foo2;
```

Comment:
So this one (and one other) is weird: the first "use of undeclared lifetime name `'test`" error is emitted in the "type_collecting" portion of `rustc_typeck::check_crate`, then the function returns because of `track_errors` and an error being emitted. Before though, the second error was still emitted? I'm confused.